src/util/url.ts

```diff
@@ -13,6 +15,32 @@ function getBaseUrl(url: URL): string {
   return baseUrl;
 }
 
+function getEndpointPath(
+  endpoint: ConfigurablePaths,
+  realmPath?: string,
+  customPaths?: CustomPathConfig,
+): string {
+  const realmUrlPath = getRealmUrlPath(realmPath);
+  const defaultPaths = {
```

Comment:
There an interface for this at `src/config/interfaces.ts`

Reply:
Being that this is an object literal, I consider it to have its own interface. I'm a big fan of type inference for literally defined objects like this.